File: RoomExercise/app/src/main/java/com/example/roomexercise/EmployeeDatabase.java
package com.example.roomexercise;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = EmployeeEntity.class,version = 2)
public abstract class EmployeeDatabase extends RoomDatabase {
    public abstract EmployeeDao employeeDao();
    private static EmployeeDatabase INSTANCE;
    public static final String DB_NAME = "employeeDB";

    public static synchronized EmployeeDatabase getInstance(Context context){
        if (INSTANCE == null){
            INSTANCE = Room.databaseBuilder(context.getApplicationContext(),EmployeeDatabase.class,DB_NAME)
                    .fallbackToDestructiveMigration()
                    .allowMainThreadQueries()
                    .build();
        }
        return INSTANCE;
    }
}

File: RoomExample/app/src/main/java/com/example/roomexample/room/EmployeeDatabase.kt
package com.example.roomexample.room

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = arrayOf(Employee::class),version = 1)
abstract class EmployeeDatabase: RoomDatabase() {
    abstract fun employeeDao(): EmployeeDao
    companion object {

        // For Singleton instantiation
        @Volatile private var instance: EmployeeDatabase? = null

        fun getInstance(context: Context): EmployeeDatabase {
            return instance
                ?: synchronized(this) {
                instance
                    ?: buildDatabase(
                        context
                    )
                        .also { instance = it }
            }
        }
        private fun buildDatabase(context: Context): EmployeeDatabase {
            return Room.databaseBuilder(context, EmployeeDatabase::class.java, "EmployeeDatabase")
                .build()
        }
    }
}